tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,272 @@
+multi_test "Setup a room, and havve the first user leave (SPEC-216)",
+
+    requires => [qw(
+        make_test_room change_room_powerlevels
+            do_request_json_for user more_users
+        can_create_room
+    )],
+
+    provides => [qw( departed_room_id )],
+
+    # User A creates a room.
```

Comment:
Woah. This is a whole lot of comments.

If the code isn't clear enough on its own, could we pull out well-named helper subs, so that the code reads more like these comments, rather than write everything both in non-compilable and compilable form?